src/common/nbkbase.cpp

```diff
@@ -197,4 +197,25 @@ void wxNotebookBase::SendPageChangedEvent(int nPageOld, int nPageNew)
     GetEventHandler()->ProcessEvent(event);
 }
 
+wxDirection wxNotebookBase::GetTabOrientation() const
+{
+    long style = GetWindowStyle();
+    if (style & wxBK_BOTTOM)
+        return wxBOTTOM;
+    else if (style & wxBK_RIGHT)
+        return wxRIGHT;
+    else if (style & wxBK_LEFT)
+        return wxLEFT;
+
+    // wxBK_TOP == 0 so we don't have to test for it
+    return wxTOP;
+}
+
+wxRect wxNotebookBase::GetTabRect(size_t WXUNUSED(n)) const
+{
+    wxFAIL_MSG(wxT("Not implemented"));
```

Comment:
Let's not use `wxT()` in the new code:
```suggestion
    wxFAIL_MSG("Not implemented");
```